ops/ops/interface_kube_control/provides.py

```diff
@@ -0,0 +1,118 @@
+import json
+from collections import namedtuple
+
+from ops import CharmBase, Relation, Unit
+from typing import List
+
+AuthRequest = namedtuple("KubeControlAuthRequest", ["unit", "user", "group"])
+
+
+class KubeControlProvides:
+    """Implemenets the Provides side of the kube-control interface."""
+
+    def __init__(self, charm: CharmBase, endpoint: str):
+        self.charm = charm
+        self.endpoint = endpoint
+
+    @property
+    def auth_requests(self) -> List[AuthRequest]:
+        requests = []
+        for relation in self.relations:
+            for unit in relation.units:
+                user = relation.data[unit].get("kubelet_user")
+                group = relation.data[unit].get("auth_group")
+                if user and group:
+                    request = AuthRequest(unit=unit.name, user=user, group=group)
+                    requests.append(request)
+        requests.sort()
+        return requests
```

Comment:
Just a suggestion; feel free to dismiss it if you find it hard to read.
```suggestion
    def auth_requests(self) -> List[AuthRequest]:
        requests = [
            AuthRequest(unit=unit.name, user=user, group=group)
            for relation in self.relations
            for unit in relation.units
            if (user := relation.data[unit].get("kubelet_user")) and (group := relation.data[unit].get("auth_group"))
        ]
        requests.sort()
        return requests
```